src/org/zaproxy/zap/extension/httpsinfo/MDialog.java

```diff
@@ -0,0 +1,319 @@
+			.getExtension(ExtensionAlert.NAME);
+
+	boolean ordered = false;
+	Map<String, Integer> versionStrings;
+	SSLServer server;
+	HttpMessage baseMessage;
+
+	JTextArea general;
+	JLabel spinnerPref;
+	JTextArea cipherSuites;
+	JComboBox<String> suppVers;
+	JButton refresh;
+
+	public MDialog(SSLServer s, HttpMessage msg) {
+		super();
+		this.server = s;
+		this.baseMessage = msg;
+		this.setIconImage(new ImageIcon(ExtensionHttpsInfo.class.getResource(ExtensionHttpsInfo.ICON_PATH)).getImage());
+		this.setTitle(Constant.messages.getString("httpsinfo.name"));
+		this.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+		Dimension d = Toolkit.getDefaultToolkit().getScreenSize();
+		int x = (d.width - getSize().width) / 2;
+		int y = (d.height - getSize().height) / 2;
+		this.setLocation(x, y);
+		init();
+		addComponentsToPane(getContentPane());
+		this.pack();
+		updateContent();
+		this.setVisible(true);
+	}
+
+	public void init() {
+		general = new JTextArea();
+		general.setEditable(false);
+		spinnerPref = new JLabel(Constant.messages.getString("httpsinfo.spinnerpref.label"));
+		cipherSuites = new JTextArea();
+		cipherSuites.setEditable(false);
+		refresh = new JButton(Constant.messages.getString("httpsinfo.serverpref.button"));
+		refresh.addActionListener(new ActionListener() {
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				new Thread(new Runnable() {
+					@Override
+					public void run() {
+						refresh.setEnabled(false);
+						server.orderCertificates();
+						ordered = true;
+						refresh.setVisible(false);
+						updateContent();
+					}
+				}).start();
+			}
+		});
+		versionStrings = new HashMap<String, Integer>();
+		for (int v : server.getSupportedVersions()) {
+			versionStrings.put(SSLServer.versionString(v), v);
+		}
+		Object[] list = versionStrings.keySet().toArray();
+		String[] slist = new String[list.length];
+		for (int i = 0; i < slist.length; i++) {
+			slist[i] = list[i].toString();
+		}
+		Arrays.sort(slist);
+		suppVers = new JComboBox<String>(slist);
+		suppVers.addActionListener(new ActionListener() {
+			@Override
+			public void actionPerformed(ActionEvent arg0) {
+				showCipherSuites();
+
+			}
+		});
+	}
+
+	public void updateContent() {
+		showGeneral();
+		showCipherSuites();
+	}
+
+	public void showGeneral() {
+		StringBuffer content = new StringBuffer(
+				Constant.messages.getString("httpsinfo.general.server.leadin", server.getServerName()));
+		if (server.getCert().size() == 0) {
+			content.append(Constant.messages.getString("httpsinfo.general.cert.notfound"));
+			new Thread(new Runnable() {
+				@Override
+				public void run() {
+					server.updateCertificate();
+					showGeneral();
+				}
+			}).start();
+		} else {
+			content.append(Constant.messages.getString("httpsinfo.general.cert.heading"));
+			for (String cc : server.getCert()) {
+				content.append("  " + cc + NEWLINE);
```

Comment:
The string concatenations should be replaced with append calls, otherwise it creates other StringBuilders. (same comment to other concatenations)